### PR TITLE
New clipboard

### PR DESCRIPTION
**Pull request by vxsx**
also fixes important modal bug

there's an unsolved problem with cancelling while dragging and z-index issue
but overall it's gonna be pretty much like this

screen shot 2015-09-21 at 18 08 48

P.S. ignore the shrinkwrap, i had to update the jscs

### CODE REVIEW

File: .jscsrc [hunk 1]
```diff
@@ -6,7 +6,11 @@
         "allowRegex": true,
         "allowUrlComments": true
     },
+    "requireDotNotation": {
+        "allExcept": [ "keywords" ]
```
Comment (FinalAngel) [inline .jscsrc]: why the extra space here?

File: cms/static/cms/js/modules/cms.clipboard.js [hunk 2]
```diff
@@ -33,7 +43,7 @@
                 this.triggerRemove = this.clipboard.find('.cms-clipboard-empty a');
 
                 // states
-                this.click = (document.ontouchstart !== null) ? 'click.cms' : 'touchend.cms click.cms';
+                this.click = 'click.cms';
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.clipboard.js]: click.cms.clipboard?

File: cms/static/cms/js/modules/cms.clipboard.js [hunk 3]
```diff
@@ -47,46 +57,35 @@
...
-                        return false;
-                    }
-
-                    // show element
-                    that.containers.stop().css({ 'margin-left': -that.options.position }).data('open', false);
-                    el.stop().animate({ 'margin-left': 0 }, that.options.speed);
-                    el.data('open', true);
+                // FIXME kind of a magic number for 1 item in clipboard
+                var minHeight = 113;
+                var pluginsList = this.clipboard.find('.cms-clipboard-containers');
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.clipboard.js]: this is not yet `this.ui` based or?

File: cms/static/cms/js/modules/cms.modal.js [hunk 4]
```diff
@@ -176,6 +173,7 @@ var CMS = window.CMS || {};
              * @param [opts.height] {Number} sets the height of the modal
              */
             open: function open(opts) {
+                this._events();
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.modal.js]: I'd add some comments here to make the positioning clear

File: cms/static/cms/js/modules/cms.modal.js [hunk 5]
```diff
@@ -246,6 +255,8 @@ var CMS = window.CMS || {};
                     });
                 }
 
+                this.trigger('cms.modal.loaded');
```
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.modal.js]: wouldn't you need the above ui.modal.trigger as well?

File: cms/static/cms/sass/components/_clipboard.scss [hunk 6]
```diff
@@ -119,15 +86,61 @@
...
+        .cms-dragarea {
+            padding-top: $padding-base;
+        }
+        .cms-draggable {
+            position: relative;
+            top: 0;
+            left: 0;
+            z-index: z(clipboard, draggable);
+            border-radius: $border-radius-base;
+            cursor: move;
```
Comment (FinalAngel) [inline cms/static/cms/sass/components/_clipboard.scss]: positioning (check all)

File: cms/templates/cms/toolbar/dragitem.html [hunk 7]
```diff
@@ -33,12 +33,14 @@
                 {% extra_menu_items plugin %}
             </div>
         </div>
+        {% if not clipboard %}
         <div class="cms-plugin-picker" data-touch-action="pan-y">
```
Comment (FinalAngel) [inline cms/templates/cms/toolbar/dragitem.html]: indent

File: cms/static/cms/js/modules/cms.clipboard.js (hunk 3 above)
Comment (vxsx) [inline cms/static/cms/js/modules/cms.clipboard.js]: nope

Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.clipboard.js]: :crying_cat_face:

File: cms/static/cms/js/modules/cms.modal.js (hunk 5 above)
Comment (vxsx) [inline cms/static/cms/js/modules/cms.modal.js]: not really

**Comment by FinalAngel** [comment]
LGTM

**Pull request merged**